components/org.wso2.carbon.metrics.data.service/src/main/java/org/wso2/carbon/metrics/data/service/MetricsDataService.java

```diff
@@ -191,6 +185,37 @@ private long getStartTime(String from) {
         return sourcesList.toArray(new String[sourcesList.size()]);
     }
 
+    public MetricHierarchyData getHierarchy(String source, String path) {
+        Map<String, MetricType> hierarchicalMetrics = reporterDAO.queryHierarchicalMetrics(source, path);
+        Set<String> childrenNames = new TreeSet<>();
+        List<MetricMeta> metrics = new ArrayList<>();
+        path = (path != null && !path.isEmpty()) ? path : "";
```

Comment:
@nirmal070125 thanks for pointing out. However, currently we don't commons-lang3 as a dependency.